src/converters/table-cell-refresh-post-fixer.js

```diff
@@ -39,3 +39,46 @@ function tableCellRefreshPostFixer( model ) {
 
 	return fixed;
 }
+
+// Check if a table cell in the view requires refreshing.
+//
+// This methods detects changes that will require:
+// - <span> to <p> rename in the view,
+// - adding a missing <paragraph>,
+// - or wrapping a text node in <paragraph>
+//
+// thus requiring refreshing the table cell view.
+//
+// @param {module:engine/model/element~Element} tableCell Table cell to check.
+// @param {String} type Type of change.
+function checkRefresh( tableCell, type ) {
+	// If children of a table cell were removed - refresh it.
+	if ( !tableCell.childCount ) {
+		return true;
+	}
+
+	const children = Array.from( tableCell.getChildren() );
+	const hasInnerText = children.some( child => child.is( 'text' ) );
+
+	// If a bare text node (not wrapped in a paragraph) was added - refresh it.
+	if ( hasInnerText ) {
+		return true;
+	}
+
+	const hasInnerParagraph = children.some( child => child.is( 'paragraph' ) );
+
+	// If there is no paragraph in table cell then the view doesn't require refreshing.
+	if ( !hasInnerParagraph ) {
```

Comment:
What if there was a clean paragraph (`<span>`) and it was replaced by an image?

Reply:
No need to refresh.